### DISCUSSION TITLE
How to increase the code location load (and reload) timeouts

### DISCUSSION THREAD

**Discussion by abi-mutinex** [dagster-io/dagster#18723] [Q&A] [answered]
After upgrading to Dagster 1.5.5, we've seen a high number of code locations getting timed out after a deployment. I know for a fact that they almost all of them finish under 60 seconds with a couple taking 100 seconds or so. 

It has never taken 300s but this error suggests that it does.

`Exception: Timed out waiting for gRPC server to start after 300s with arguments: "/usr/local/bin/python -m dagster api grpc --lazy-load-user-code --socket /tmp/tmpx2gtadjm --heartbeat --heartbeat-timeout 30 --fixed-server-id 8e96909e-f2ee-46f3-8f56-f8dec813346b --log-level info --location-name <redacted> --container-image <redacted> --container-context {"k8s": {"env": [{"name": "ENV", "value": "staging"}, {"name": "DAGSTER_GRPC_TIMEOUT_SECONDS", "value": "300"}, {"name": "MTX_ORGANIZATION_ID", "value": "47"}], "env_config_maps": <redacted>, "image_pull_policy": "Always", "namespace": "dagster", "service_account_name": "dagster-user-deployments-sa"}} -f spina.py -d /app". Most recent connection error: dagster._core.errors.DagsterUserCodeUnreachableError: Could not reach user code server. gRPC Error code: UNAVAILABLE`

As you can see below, we have put 3 configs in place to avoid this problem but I don't think the 300s timeout is being honoured.

```
{
  "name": "abc-30",
  "image": {
    "repository": "abc-repo",
    "tag": "9c271a38950260da77591af84ab19d531064c49e",
    "pullPolicy": "Always"
  },
  "codeServerArgs": [
    "--python-file",
    "abc.py",
    "--startup-timeout",
    "300"
  ],
  "port": 3030,
  "envConfigMaps": [
    {
      "name": "abc-configmap"
    }
  ],
  "env": [
    {
      "name": "ENV",
      "value": "__MTX_ENV"
    },
    {
      "name": "DAGSTER_GRPC_TIMEOUT_SECONDS",
      "value": "300"
    },
    {
      "name": "abc_id",
      "value": "30"
    }
  ],
  "additionalInstanceConfig": {
    "code_servers": {
      "reload_timeout": "300"
    }
  }
}

```

Could you help please?

**Comment by gibsondan** [comment, marked as answer]
It looks like you have the timeout set to 300 seconds, and the error message suggests that the timeout it is using is indeed 300 seconds. You should be able to increase or decrease the timeout by changing that —startup-timeout argument, is that different than what you are seeing? 

If it’s not using 300 seconds and the error message is inaccurate - how many seconds does it take after reloading before that error message shows up?

If the question is instead “why is it timing out after 300 seconds” rather than “how do I change the configuration”, the tools I would use to try to answer that are adding logs in the startup process to see what it’s doing, or using “ps” and “py-spy” on the container to see it the subprocess is actually starting up and what it’s doing during those 300 seconds (see https://github.com/dagster-io/dagster/discussions/14771 for more information on how to use py-spy).

 I’d also check the memory usage on the pod to see if the subprocess might be OOMing - it’s possible it’s timing out due to a crash.
